odoo/addons/test_new_api/tests/test_properties.py

```diff
@@ -1623,3 +1625,86 @@ def test_properties_field_search_unaccent(self):
         self.assertNotIn(self.message_1, result)
         self.assertNotIn(self.message_2, result)
+
+    @mute_logger('odoo.fields')
+    def test_properties_field_search_orderby_string(self):
+        """Test that we can order record by properties string values."""
+        (self.message_1 | self.message_2 | self.message_3).discussion = self.discussion_1
+        self.message_1.attributes = [{
+            'name': 'mychar',
+            'type': 'char',
+            'value': 'BB',
+            'definition_changed': True,
+        }]
+        self.message_2.attributes = {'mychar': 'AA'}
+        self.message_3.attributes = {'mychar': 'CC'}
+
+        self.env.flush_all()
+
+        result = self.env['test_new_api.message'].search(
+            domain=[['attributes.mychar', '!=', False]],
+            order='attributes.mychar ASC')
+        self.assertEqual(result[0], self.message_2)
+        self.assertEqual(result[1], self.message_1)
+        self.assertEqual(result[2], self.message_3)
+
+        result = self.env['test_new_api.message'].search(
+            domain=[['attributes.mychar', '!=', False]],
+            order='attributes.mychar DESC')
+        self.assertEqual(result[0], self.message_3)
+        self.assertEqual(result[1], self.message_1)
+        self.assertEqual(result[2], self.message_2)
+
+    @mute_logger('odoo.fields')
+    def test_properties_field_search_orderby_integer(self):
+        """Test that we can order record by properties integer values."""
+        (self.message_1 | self.message_2 | self.message_3).discussion = self.discussion_1
+        self.message_1.attributes = [{
+            'name': 'myinteger',
+            'type': 'integer',
+            'value': 22,
+            'definition_changed': True,
+        }]
+        self.message_2.attributes = {'myinteger': 111}
+        self.message_3.attributes = {'myinteger': 33}
+
+        self.env.flush_all()
+
+        result = self.env['test_new_api.message'].search(
+            domain=[['attributes.myinteger', '!=', False]],
+            order='attributes.myinteger ASC')
+        self.assertEqual(result[0], self.message_1)
+        self.assertEqual(result[1], self.message_3)
+        self.assertEqual(result[2], self.message_2)
+
+        result = self.env['test_new_api.message'].search(
+            domain=[['attributes.myinteger', '!=', False]],
+            order='attributes.myinteger DESC')
+        self.assertEqual(result[0], self.message_2)
+        self.assertEqual(result[1], self.message_3)
+        self.assertEqual(result[2], self.message_1)
+
+    @mute_logger('odoo.fields')
+    def test_properties_field_search_orderby_injection(self):
+        """Check the restriction on the property name."""
+        self.message_1.attributes = [{
+            'name': 'myinteger',
+            'type': 'integer',
+            'value': 22,
+            'definition_changed': True,
+        }]
+
+        for c in '! ()"\'.':
```

Comment:
See my comment about making the check "earlier", as I think we are duplicating these tests.